demo-simple/src/main/java/com/novoda/downloadmanager/demo/MainActivity.java

```diff
@@ -124,7 +129,33 @@ protected void onCreate(Bundle savedInstanceState) {
         databaseMigrationUpdates.setText(migrationStatus.status().toRawValue());
     };
 
-    private final View.OnClickListener startMigrationOnClick = v -> downloadMigrator.startMigration("migrationJob", getDatabasePath("downloads.db"), V1_BASE_PATH);
+    private final View.OnClickListener startMigrationOnClick = v -> {
```

Comment:
Where all the magic happens. We extract the migrations from the v1 database and add to the wrapper object `VersionOnePartialDownloadBatch` which we use to perform the `download` and then delete the records from the v1 database. This will be wrapped in a runnable or some other thing later and moved off of the main-thread.